api/src/main/java/org/pmiops/workbench/db/model/BillingProjectBufferEntry.java

```diff
@@ -84,7 +95,9 @@ public BillingProjectBufferStatus getStatusEnum() {
     return StorageEnums.billingProjectBufferStatusFromStorage(status);
   }
 
-  public void setStatusEnum(BillingProjectBufferStatus status) {
+  public void setStatusEnum(
+      BillingProjectBufferStatus status, Supplier<Timestamp> currentTimestamp) {
```

Comment:
This is definitely on the fence for me. We don't really have business logic in these model classes anywhere else in the code AFAIK, but it also makes all the call sites pretty clean and ensures we don't miss any. Of course, someone could go straight to the `setStatus()` method and bypass this.

Reply:
Yeah... I wasn't sure about this either but I decided the protection against misuse is worth it. `setStatus()` and `setLastStatusChangedTime()` are actually both private so the only way to update either is through this method.